test/NHSD.BuyingCatalogue.Documents.API.Tests/AzureBlobStorageTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using FluentAssertions;
+using Moq;
+using NHSD.BuyingCatalogue.Documents.API.Storage;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Documents.API.Tests
+{
+    [TestFixture]
+    internal class AzureBlobStorageTests
+    {
+        private Mock<BlobContainerClient> _blobContainerClientMock;
+        private Mock<AsyncPageable<BlobItem>> _blobPageMock;
+        private IEnumerable<BlobItem> _blobItems;
+        private AzureBlobStorage _storage;
+
+        [SetUp]
+        public void Setup()
+        {
```

Comment:
Should we take this opportunity to make the tests run in parallel or is it not worth it yet? We would need a test context to encapsulate this logic with the member variables above.

Reply:
@sam-roberts-bjss I would prefer that we avoid setup methods and test context as much as possible: i.e. each test is responsible for its own setup (no shared state means the tests are inherently parallelizable).

However, it's probably not worth it yet (currently only a small number of tests that run quickly).

Reply:
If we are to make the tests run in parallel this will be useful. With it such a small project it could be used as a testing ground to potentially bring the format into the Buying Catalogue API. I think we should create a tech debt task.